lib/pages/homeserver_picker/homeserver_picker.dart

```diff
@@ -158,14 +157,24 @@ class HomeserverPickerController extends State<HomeserverPicker> with ConnectPag
         matrix.loginRegistrationSupported = e.requireAdditionalAuthentication;
       }
 
-      if (!ssoSupported && matrix.loginRegistrationSupported == false) {
-        // Server does not support SSO or registration. We can skip to login page:
-        VRouter.of(context).to('login');
-      } else if (ssoSupported && matrix.loginRegistrationSupported == false) {
+      if (ssoSupported && matrix.loginRegistrationSupported == false) {
         setState(() {
           state = HomeserverState.ssoLoginServer;
           FocusManager.instance.primaryFocus?.unfocus();
         });
+        Map<String, dynamic>? rawLoginTypes;
```

Comment:
```suggestion
        final rawLoginTypes = await Matrix.of(context)
```

Reply:
You wrongly, `rawLoginTypes` is response of `Matrix.of(context).getLoginClient().request...`